CodeStarterCamp_Week2/main.swift

```diff
@@ -8,23 +8,39 @@
 
 import Foundation
 
-let myLottoNumbers: Set<Int> =  [1, 2, 3, 4, 5, 6]
-var numberfull: [Int] = Array<Int>(1...45)
-var lotto: Set<Int> = Set<Int>()
+var lottoDictionary = [String : Set<Int>]()
+var numberFull: [Int] = Array<Int>(1...45)
+var lottoNumber: Set<Int> = Set<Int>()
 
-func pickRandomLottoNumber() {
-    while lotto.count < 6 {
-        lotto.insert(Int.random(in: 1...45))
+func pickRandomLottoNumber() -> Set<Int> {
+    var lottoNumber: Set<Int> = Set<Int>()
+    while lottoNumber.count < 6 {
+        lottoNumber.insert(Int.random(in: 1...45))
     }
+    return lottoNumber
 }
 
-func matchingLotto(myLottoNumbers: Set<Int>, lotto: Set<Int>) {
-    let result: Set<Int> = myLottoNumbers.intersection(lotto)
-    if result.isEmpty {
-        print("아쉽지만 겹치는 번호는 없습니다.")
-        return
+func storeLottoNumber(round: Int) {
+    lottoDictionary["\(round)회차"] = pickRandomLottoNumber()
+}
+
+func getLottoNumber() {
+    for i in 1...5 {
+        storeLottoNumber(round: i)
+    }
+}
+
+func searchLottoNumber(times: Int) {
+    if let lottoOptional = lottoDictionary["\(times)회차"] {
+        print("\(times)회차의 로또 당첨 번호는 \(lottoOptional.map { String($0) }.joined(separator: ", ")) 입니다.")
+    } else {
+        print("\(times)회차는 존재하지 않습니다.")
```

Comment:
`if let` 사용 너무 좋습니다!!

다만 저 같은경우에는 else의 사용을 지양하는 편인데요. 왜 지양해야하는 지에 대해서는 한 번 찾아보시는 것도 좋을 것 같아요!!




```suggestion
func searchLottoNumber(times: Int) {
    if let lottoOptional = lottoDictionary["\(times)회차"] {
        return print("\(times)회차의 로또 당첨 번호는 \(lottoOptional.map { String($0) }.joined(separator: ", ")) 입니다.")
    } 
    return print("\(times)회차는 존재하지 않습니다.")
```

Reply:
if else 와 if return은 복잡한 중첩을 없애는 가독성의 차이인 것으로 확인되는데, 다르게 생각하시는 이유가 있을까요?

Reply:
가독성의 차이도 있지만 if문이 많아지면 많아질 수록 그에 따른 많은 단점들이 생길 수 있을것 같아요.
[Else if와 Else를 피하고 Early Return하자](https://velog.io/@jangws/9.-Else-if와-Else를-피하고-Early-Return하자) 가볍게 한번 읽어보시는걸 추천드릴게요!